server/svix-server/src/v1/endpoints/endpoint/crud.rs

```diff
@@ -125,36 +125,60 @@ pub(super) async fn update_endpoint(
     Path((_app_id, endp_id)): Path<(ApplicationIdOrUid, EndpointIdOrUid)>,
     ValidatedJson(data): ValidatedJson<EndpointIn>,
     AuthenticatedApplication { permissions, app }: AuthenticatedApplication,
-) -> Result<Json<EndpointOut>> {
+) -> Result<(StatusCode, Json<EndpointOut>)> {
     let endp = endpoint::Entity::secure_find_by_id_or_uid(app.id.clone(), endp_id)
         .one(db)
-        .await?
-        .ok_or_else(|| HttpError::not_found(None, None))?;
+        .await?;
 
     if let Some(ref event_types_ids) = data.event_types_ids {
         validate_event_types(db, event_types_ids, &permissions.org_id).await?;
     }
     validate_endpoint_url(&data.url, cfg.endpoint_https_only)?;
 
-    let mut endp: endpoint::ActiveModel = endp.into();
-    data.update_model(&mut endp);
+    let app_uid = app.uid;
 
-    let ret = endp.update(db).await?;
+    let (status_code, ret) = match endp {
+        Some(endp) => {
+            let mut endp: endpoint::ActiveModel = endp.into();
+            data.update_model(&mut endp);
+            let ret = endp.update(db).await?;
+
+            (StatusCode::OK, ret)
+        }
+        None => {
+            let ret = endpoint::ActiveModel {
+                app_id: Set(app.id),
+                key: Set(generate_secret(
+                    &cfg.encryption,
+                    &cfg.default_signature_type,
+                )?),
```

Comment:
This is wrong, see the create method... You are always overriding the key, you should support using it.

I think you may have done it because EndpointUpdate doesn't have key? Though it's what I said on Slack, you need to change it to use EndpointIn.

At this point I'm thinking: why not just call the "create endpoint" function directly from here instead of duplicating it wrong?